core/src/main/java/org/bitcoinj/core/VersionedChecksummedBytes.java

```diff
@@ -33,29 +36,20 @@
  * <p>and the result is then Base58 encoded. This format is used for addresses, and private keys exported using the
  * dumpprivkey command.</p>
  */
-public class VersionedChecksummedBytes implements Serializable, Cloneable, Comparable<VersionedChecksummedBytes> {
-    protected final int version;
```

Comment:
Why dropping the `version` field? It's a loss of information and cannot be recreated from the network parameters alone.

Reply:
Another "minor" use case is with Litecoin, where they have P2SH addresses with header byte of `5` or `50`.

Reply:
I can add it back of course, but for the current classes it's not needed. The only information it carries is the network (use `VersionChecksummedBytes.params`) and the usecase (inspect the type of your class if you don't already know it).

What's the difference between 5 and 50 on Litecoin?

Reply:
There is no difference except do distinguish it visually

Reply:
Another issue that I see is in `SegWitAddress` where we need to store a version but we do it in a "hacky" way by using the internal `bytes`.